packages/sdk/src/modules/Flywheel.ts

```diff
@@ -221,16 +221,15 @@ export function withFlywheel<TBase extends FuseBaseConstructorWithCreateContract
         this.signer
       ) as MidasFlywheel__factory;
       const addressOfSigner = await this.signer.getAddress();
-      const mfw = await midasFlywheel.deploy();
-      const flywheelCoreInstance = this.createMidasFlywheel(mfw.address);
-      const flywheelCoreInstanceWithSigner = flywheelCoreInstance.connect(this.signer);
-      await flywheelCoreInstanceWithSigner.initialize(
+      const flywheelCore = await midasFlywheel.deploy();
+      const initializeTx = await flywheelCore.initialize(
         rewardTokenAddress,
         options?.rewardsAddress || constants.AddressZero,
         options?.boosterAddress || constants.AddressZero,
         options?.ownerAddress || addressOfSigner
       );
-      return flywheelCoreInstance;
+      await initializeTx.wait();
```

Comment:
just fyi @carlomazzaferro as you did the change. We missed to wait for the `initialize()` call to be successfull before continueing with the rest of the flywheel transactions. It worked on polygon as it was faster than the person clicked but that was not the case on BSC.